fuel/app/classes/materia/widget/instance/manager.php

```diff
@@ -101,4 +102,49 @@ public static function lock($inst_id)
 		// true if the lock is mine
 		return $locked_by == $me;
 	}
+
+	/**
+	 * Gets all widget instances related to a given input, including id or name. 
+	 * 
+	 * @param input search input
+	 * 
+	 * @return array of widget instances related to the given input
+	 */
+	public static function get_search($input)
+	{
+		$results = \DB::select()
+			->from('widget_instance')
+			->where('id', 'LIKE', "%$input%")
+			->or_where('name', 'LIKE', "%$input%")
+			->order_by('created_at', 'desc')
+			->execute()
+			->as_array();
+
+		$instances = [];
+		foreach ($results as $r)
+		{
+			$widget = new Widget();
+			$widget->get($r['widget_id']);
+			$inst = new Widget_Instance([
+				'id'              => $r['id'],
+				'user_id'         => $r['user_id'],
+				'name'            => $r['name'],
+				'is_student_made' => (bool) $r['is_student_made'],
+				'student_access'  => Perm_Manager::accessible_by_students($r['id'], Perm::INSTANCE),
+				'guest_access'    => (bool) $r['guest_access'],
+				'is_draft'        => (bool) $r['is_draft'],
+				'created_at'      => $r['created_at'],
+				'open_at'         => $r['open_at'],
+				'close_at'        => $r['close_at'],
+				'attempts'        => $r['attempts'],
+				'is_deleted'			=> (bool) $r['is_deleted'],
+				'embedded_only'   => (bool) $r['embedded_only'],
+				'widget'          => $widget,
+			]);
```

Comment:
Move Perm_Manager::accessible_by_students before creation to clean up.
fix alignments

```suggestion
			$widget->get($r['widget_id']);
			$student_access = Perm_Manager::accessible_by_students($r['id'], Perm::INSTANCE);
			$inst = new Widget_Instance([
				'id'              => $r['id'],
				'user_id'         => $r['user_id'],
				'name'            => $r['name'],
				'is_student_made' => (bool) $r['is_student_made'],
				'student_access'  => $student_access,
				'guest_access'    => (bool) $r['guest_access'],
				'is_draft'        => (bool) $r['is_draft'],
				'created_at'      => $r['created_at'],
				'open_at'         => $r['open_at'],
				'close_at'        => $r['close_at'],
				'attempts'        => $r['attempts'],
				'is_deleted'      => (bool) $r['is_deleted'],
				'embedded_only'   => (bool) $r['embedded_only'],
				'widget'          => $widget,
			]);
```